dashboard/src/actions/charts.ts

```diff
@@ -90,15 +93,19 @@ export function fetchCharts(
   repos: string,
   page: number,
   size: number,
+  records: Map<number, boolean>,
   query?: string,
 ): ThunkAction<Promise<void>, IStoreState, null, ChartsAction> {
   return async dispatch => {
-    dispatch(requestCharts(query));
-    try {
-      const charts = await Chart.fetchCharts(cluster, namespace, repos, page, size, query);
-      dispatch(receiveCharts(charts));
-    } catch (e) {
-      dispatch(errorChart(new FetchError(e.message)));
+    const latestElement = Array.from(records.keys()).pop() || 1;
+    if (records.get(page) === false && page <= latestElement) {
```

Comment:
It is just an extra condition for my unfruitful attempts to avoid the problems of duplicating requests.
The rationale behind this is `latestElement` is being set in `receiveCharts`: here we set to `false` the next page that should be requested (otherwise, it should be `undefined`).

So, if is properly processes page 6, it will set `7: false`. During the next request, `(records.get(7) === false && 7 <= 7)`

If something is wrong, the map could possibly include `7:false, 8:false` , the `latestElement` would be 8 whereas `page` is still 7. 

I'm sure there exists a better solution for the root cause.

Regarding the renaming, it's not the `lastPage`, but the `lastPendingPage`, but I'm ok with whichever name.